core/src/main/java/org/infinispan/container/offheap/OffHeapEntryFactoryImpl.java

```diff
@@ -359,6 +360,55 @@ public boolean equalsKey(long address, WrappedBytes wrappedBytes) {
       return true;
    }
 
+   /**
+    * Returns whether entry is expired or not.
+    * @param address the address of the entry's key to check
+    * @return {@code true} if the entry is expired, {@code false} otherwise
+    */
+   @Override
+   public boolean isExpired(long address) {
+      // 16 bytes for eviction if needed (optional)
+      // 8 bytes for linked pointer
+      int offset = evictionEnabled ? 24 : 8;
+
+      byte metadataType = MEMORY.getByte(address, offset);
+      offset += 1;
+      // hashCode
+      offset += 4;
+      // key length
+      int keyLenght = MEMORY.getInt(address, offset);
+      offset += 4;
+      // value length
+      offset += 4;
+      // key
+      offset += keyLenght;
+
+      byte[] metadataBytes;
+      long now = timeService.wallClockTime();
+      switch (metadataType) {
```

Comment:
This isn't quite correct either as we could have VERSION or CUSTOM flag set on the type in which case this switch statement wouldn't work properly.

https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/container/offheap/OffHeapEntryFactoryImpl.java#L278